contracts/citycoin.clar

```diff
@@ -137,6 +139,44 @@
 ;; The fungible token that can be Stacked.
 (define-fungible-token citycoins)
 
+(define-constant MINING-ACTIVATION-THRESHOLD u1)  ;; how many miners have to register to kickoff countdown to mining activation
+(define-constant MINING-ACTIVATION-DELAY u100)   ;; how many blocks after last miner registration mining will be activated   
+
+(define-data-var signalingMinersNonce uint u0)
+
+(define-map SignalingMiners
+    { miner: principal }
+    { id: uint }
+)
+
+(define-public (register-miner)
+    (let
+        (
+            (newId (+ u1 (var-get signalingMinersNonce)))
+        )
+        (asserts! (is-none (map-get? SignalingMiners {miner: tx-sender}))
+            (err ERR-MINER-ALREADY-REGISTERED))
+
+        (asserts! (<= newId MINING-ACTIVATION-THRESHOLD)
+            (err ERR-MINING-ACTIVATION-THRESHOLD-REACHED))
+        
+        (map-set SignalingMiners
+            {miner: tx-sender}
+            {id: newId}
+        )
+        
+        (var-set signalingMinersNonce newId)
+
+        (if (is-eq newId MINING-ACTIVATION-THRESHOLD) 
+            (begin
+                (var-set first-stacking-block (+ block-height MINING-ACTIVATION-DELAY))
```

Comment:
Given `FIRST_STACKING_BLOCK` is a constant, should this be `firstStackingBlock` following the naming conventions for variables? Or should it be changed so its no longer a constant?

Reply:
All existing variables, maps, functions, parameters etc. follows naming convention chosen by @jcnelson and I didn't want to change any of that as this would be a very big change.

Now that you brought up this topic, I think I made a mistake and I should stick to his convention rigorously. Introducing new standards like in example below is probably more confusing than beneficial.
```
(define-data-var signalingMinersNonce uint u0)

(define-map SignalingMiners
    { miner: principal }
    { id: uint }
)
```
I'll change it to all lower case, dash spaced style to keep the code as consistent as possible.

Reply:
Consistency is 🔑 - I'll note this for my changes as well!